### ISSUE TITLE
BIRTReports Exceptions and Errors

### ISSUE THREAD

**Issue by kingthorin** [zaproxy/zaproxy#2235]
Based on : https://raw.githubusercontent.com/zapbot/zap-mgmt-scripts/6cea54f00559b49fd9f994bf1c68e64eb1c74e65/reports/wavsep-1.5-weekly-All-M-M.logs.txt

> 23503 [ZAP-daemon] WARN org.zaproxy.zap.control.BaseZapAddOnXmlData  - Ignoring empty "ascanrule" entry in add-on "birtreports".
> 23504 [ZAP-daemon] WARN org.zaproxy.zap.control.BaseZapAddOnXmlData  - Ignoring empty "pscanrule" entry in add-on "birtreports".
> 23504 [ZAP-daemon] WARN org.zaproxy.zap.control.BaseZapAddOnXmlData  - Ignoring empty "file" entry in add-on "birtreports".
> 23504 [ZAP-daemon] INFO org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate  - Installing new addon birtreports v2
> 23512 [ZAP-daemon] ERROR org.zaproxy.zap.control.AddOnLoaderUtils  - Declared "extension" was not found: org.zaproxy.zap.extension.birtreports
> java.lang.ClassNotFoundException
>   at org.zaproxy.zap.control.AddOnClassLoader.findClass(AddOnClassLoader.java:213)
>   at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
>   at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
>   at org.zaproxy.zap.control.AddOnLoaderUtils.loadAndInstantiateClassImpl(AddOnLoaderUtils.java:93)
>   at org.zaproxy.zap.control.AddOnLoaderUtils.loadAndInstantiateClass(AddOnLoaderUtils.java:68)
>   at org.zaproxy.zap.control.AddOnLoader.loadAddOnExtension(AddOnLoader.java:722)
>   at org.zaproxy.zap.control.AddOnLoader.loadAddOnExtensions(AddOnLoader.java:713)
>   at org.zaproxy.zap.control.AddOnLoader.getExtensions(AddOnLoader.java:679)
>   at org.zaproxy.zap.control.ExtensionFactory.loadAddOnExtensions(ExtensionFactory.java:236)
>   at org.zaproxy.zap.control.AddOnInstaller.installAddOnExtensions(AddOnInstaller.java:177)
>   at org.zaproxy.zap.control.AddOnInstaller.install(AddOnInstaller.java:86)
>   at org.zaproxy.zap.control.AddOnLoader.addAddOnImpl(AddOnLoader.java:368)
>   at org.zaproxy.zap.control.AddOnLoader.addAddon(AddOnLoader.java:344)
>   at org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate.install(ExtensionAutoUpdate.java:1004)
>   at org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate.installNewExtensions(ExtensionAutoUpdate.java:470)
>   at org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate.waitAndInstallDownloads(ExtensionAutoUpdate.java:1732)
>   at org.zaproxy.zap.extension.autoupdate.ExtensionAutoUpdate.execute(ExtensionAutoUpdate.java:1597)
>   at org.parosproxy.paros.extension.ExtensionLoader.runCommandLine(ExtensionLoader.java:386)
>   at org.parosproxy.paros.control.Control.runCommandLine(Control.java:302)
>   at org.zaproxy.zap.DaemonBootstrap$1.run(DaemonBootstrap.java:90)
>   at java.lang.Thread.run(Thread.java:745)

**Comment by kingthorin** [comment]
The issues above proved very easy to address with changes to ZapAddOn.xml. Unfortunately addressing those lead to further and varying problems including incorrect use of i18n functionality, and failure for the reporting engine to properly load the necessary classes it depends on, etc.

Attached is a stack trace I received once I address the original issue above as well as some of the i18n issues. Unfortunately further fixes are not a priority at this time. Moving this issues to Low priority, un-assigning @kingthorin.

As agreed via email this addon should be removed from the market place until further and full corrections can take place.
[birt_stack_20160213.txt](https://github.com/zaproxy/zaproxy/files/159633/birt_stack_20160213.txt)

**Comment by thc202** [comment]
Didn't mean to close it.... anyway, the add-on is no longer available in the marketplace.

**Comment by kingthorin** [comment]
Depends on: https://github.com/zaproxy/zaproxy/issues/2619

**Comment by thc202** [comment]
Not really, it can still be fixed without that issue (it does make the fix easier, though).
